server/src/main/java/io/crate/planner/statement/DeletePlanner.java

```diff
@@ -104,7 +105,7 @@ private static Plan planDelete(AnalyzedDeleteStatement delete, PlannerContext co
             return new DeleteById(tableRel.tableInfo(), detailedQuery.docKeys().get());
         }
         if (table.isPartitioned() && query instanceof Input<?> input && DataTypes.BOOLEAN.sanitizeValue(input.value())) {
-            return new DeleteAllPartitions(Lists.map(table.getPartitionNames(context.clusterState().metadata()), PartitionName::asIndexName));
+            return new DeleteAllPartitions(table.ident());
```

Comment:
Could be a drop table instead?
```suggestion
            return new DropTablePlan(new AnalyzedDropTable(true, table.ident()));
```

Reply:
> Could be a drop table instead?

Nope as the table itself must stay.